src/Compilers/Core/Portable/Syntax/GreenNode.cs

```diff
@@ -195,32 +195,95 @@ internal Syntax.InternalSyntax.ChildSyntaxList ChildNodesAndTokens()
         /// <summary>
         /// Enumerates all nodes of the tree rooted by this node (including this node).
         /// </summary>
-        internal IEnumerable<GreenNode> EnumerateNodes()
+        internal EnumerateNodesEnumerable EnumerateNodes()
+            => new(this);
+
+        public readonly struct EnumerateNodesEnumerable
```

Comment:
❔ Does this need to declare that it implements `IDisposable`?

Reply:
will make IDisposable just to be sure!